packages/react-router/src/route.ts

```diff
@@ -165,11 +190,11 @@ export type BaseRouteOptions<
   TFullSearchSchema = TSearchSchema,
   TParams = {},
   TAllParams = {},
-  TRouteContextReturn = RouteContext,
-  TParentAllContext = AnyContext,
-  TAllContext = AnyContext,
   TLoaderDeps extends Record<string, any> = {},
   TLoaderDataReturn = {},
+  TParentAllContext = {},
+  TRouteContextFn = AnyContext,
```

Comment:
I'm trying out patterns to remove unnecessary type parameters. Usually we require 2 type parameters per function because we need to check the return type is `never`. If we infer the whole function, we don't need two type paramters